### ISSUE TITLE
Implement build infra around TS path aliases for convergence

### ISSUE THREAD

**Issue by Hotell** [microsoft/fluentui#16889]
# Epic

This is an Epic which will track implementation of issues pointed out in following documents:
- POC https://github.com/microsoft/fluentui/pull/16706#pullrequestreview-583015668
- RFC https://github.com/microsoft/fluentui/pull/16570#pullrequestreview-582286779

## Problem
- need to build before run/test/demo
- unified/explicit TSConfigs
- missing co-located demos via storybook

## Solution

- we'll use TSConfig path aliases as the only source of truth for converged components.

### Implementation:

**Pre-requirements**
- [x] use singleVersion policy for TypeScript (https://github.com/microsoft/fluentui/pull/16592)
- [x] use singleVersion policy for Jest (https://github.com/microsoft/fluentui/pull/16950)
- [x] use singleVersion policy for Storybook (https://github.com/microsoft/fluentui/pull/16965)

**Requirements:**
- [x] create `tsconfig.base.json` with all existing converged component paths (https://github.com/microsoft/fluentui/pull/16976)
- [x] implement TS path aliases resolution for jest (https://github.com/microsoft/fluentui/pull/17161)
       - for `npm:test` use jest binary directly instead `just-script`
- [x] api-extractor integration with TS path aliases approach (https://github.com/microsoft/fluentui/pull/17449)
- [x] new react-storybook package that doesn't depends on v8 (https://github.com/microsoft/fluentui/pull/17921)
- [x] new storybook configuration with collocated stories (https://github.com/microsoft/fluentui/pull/17866)
  - [x] implement TS path aliases resolution for webpack/storybook
       - for `npm:start` use storybook binary directly instead `just-script`
  - [x] implement "standard" storybook root config - `<monorepoRoot>/.storybook`
       - this will house common storybook configuration
  - [x] implement localised storybooks per converged package
       - `<packageFolder>/.storybook`
- ~[ ] implement root storybook app (running it will render all converged stories) `<monorepoRoot>/apps/storybook`~
       - 💡 this was applied on react-components package. Additional setup will be needed for react-components -> will be implemented in separate issue/PR
- [x] implement migration to switch all existing converged packages to this new "dev loop" behaviour | series of PR's finalised with https://github.com/microsoft/fluentui/pull/18421 
- [x] update generators to adhere to new dev flow when creating converged package https://github.com/microsoft/fluentui/pull/18421
       - should update tsconfig.base.json#paths
       - should update jest.config.js projects

## Risks (Rabbit holes)
- No idea about plop AST capabilities (might get into issues when I'll need to add new package to `monorepoRoot/jest.config.js`)
- plop is not very testable AFAIR (will need to try it out)

## Out of scope (No-gos)
- using solution TS config style (project references per package) to explicitly distinguish between environments (story/implementation/test)
- revamping all npm:scripts and how we build/publish
- use babel for transpilation
- use babel-jest for testing
- use rollup as main bundler

**Comment by JustSlone** [comment]
Update:
- 3 Pull requests out (see checklist above)
- Discovered more issues, global types are leaking, not blocking currently - TODO: Create new RFC to tackle these

**Comment by Hotell** [comment]
**Update:**
- initial path aliases are merged, react-menu is the guinea pig :)  #16976

**Comment by Hotell** [comment]
- finally resolved issues https://github.com/microsoft/fluentui/pull/17161, ready to review

March goal: finish all tasks - every conformant package has the new DX implemented

**Comment by Hotell** [comment]
- resolved review comments #17161, waiting for approval

**Comment by Hotell** [comment]
- added new task -> api-extractor integration with TS path aliases approach

> **Legend:**
> - 🛠️  = in progress
> - 👁 = ready for review
> -  🛑  = blocked
> - ✅  = done

**Status:**
- 🛠️ / api-extractor integration with TS path aliases approach
- 🛠️ / local storybooks with global config
